### DISCUSSION TITLE
How to Add Drawer with Navigation bar without bottom extra space

### DISCUSSION THREAD

**Discussion by vishal-ainow** [saadeghi/daisyui#3122] [Q&A] [answered]
As you've seen in the below URL, I'm trying to disable scrolling. When I add a drawer, the entire page scrolls because I've included a navbar (header). However, I want only the content to scroll, not the entire screen. What am I missing here?

https://play.tailwindcss.com/3aJfqd22IY?size=1048x529

**Comment by saadeghi** [comment]
Pur the navbar inside `drawer-content`

https://daisyui.com/components/drawer/#drawer-tags-structure

**Comment by vishal-ainow** [comment]
Hey @saadeghi, When I followed your approach and placed the Navbar inside `drawer-content`, it moved to the right side like [this](https://play.tailwindcss.com/kRYxqusoMO?size=1050x529). However, I need it to be displayed only at the top.

**Comment by saadeghi** [comment]
Remove `flex` class from `drawer-content`.  
When you add `flex` class, it puts items horizontally.

**Comment by vishal-ainow** [comment]
Hey @saadeghi, As you can see in the screenshot, when I remove the `flex` class from `drawer-content`, the navbar still aligns to the right side of the page. However, as indicated in the screenshot with the orange highlight, I need the navbar to fit snugly at the top. Immediately following that, the drawer should be displayed.

<img width="977" alt="Screenshot 2024-07-03 at 5 49 57 PM" src="https://github.com/saadeghi/daisyui/assets/173050928/edca1d97-8933-4bf2-8122-81b983876a27">


When I add a top bar above the drawer, I notice there's extra space at the bottom equivalent to the menu height.

<img width="989" alt="Screenshot 2024-07-03 at 6 04 21 PM" src="https://github.com/saadeghi/daisyui/assets/173050928/16943640-821a-4702-9694-efc673b21e68">


How to remove this extra space?

**Reply by saadeghi** [reply, marked as answer]
In that case, if you want the navbar on top of the sidebar as well here are the changes you need to make:
1. remove `<main class="min-h-screen">` 
2. Put back `navbar` out of `drawer`. I initially told you to put it inside, because I didn't know you want the navbar on top of the sidebar as well (thanks for the screenshot)
3. Add `h-[calc(100vh-4rem)] top-16` to `drawer-side` because your navbar is sticky and you won't want the sidebar to go under the navbar.
4. Additionally there's an issue with your menu. Remove `flex h-full flex-col overflow-auto`. You don't need any of them. Add `min-h-full` so your menu covers the height.


Here's the fixed version:

https://play.tailwindcss.com/hzj8ex41Uf?size=1088x699
